test/Microsoft.SqlTools.ServiceLayer.UnitTests/Connection/ConnectionServiceTests.cs

```diff
@@ -1210,5 +1210,14 @@ public void ConnectParamsAreInvalidIfConnectionIsNull()
             await Assert.ThrowsAsync<InvalidOperationException>(
                 () => service.GetOrOpenConnection(TestObjects.ScriptUri, ConnectionType.Query));
         }
+
+        [Fact]
+        public async Task ConnectionWithConnectionStringSucceeds()
```

Comment:
I think since it's possible to have both a connection string and the piece-meal connection properties, we should have a test for that, to make sure the expected behavior is produced.